statistics/handle/globalstats/global_stats.go

```diff
@@ -80,9 +80,15 @@ func MergePartitionStats2GlobalStats(
 	globalTableInfo *model.TableInfo,
 	isIndex bool,
 	histIDs []int64,
+	allPartitionStats map[int64]*statistics.Table,
 	getTableByPhysicalIDFn getTableByPhysicalIDFunc,
 	loadTablePartitionStatsFn loadTablePartitionStatsFunc,
```

Comment:
Can we wrap it as a closure function outside:
![image](https://github.com/pingcap/tidb/assets/7499936/27758457-a502-480f-8e99-f1bff26a9ad2)